### PR TITLE
Deprecated various functions in `skbio.util`

### LINKED ISSUE #833: Clean up skbio.util
`util only exports functions that are useful to the codebase in scikit-bio or is relevant to users within the context of biology but escapes classification otherwise`

- [ ] Remove create_dir
- [x] remove flatten
- [ ] move is_casava_v180_or_later into fastq sniffer
- [ ] remove remove_files

### PR DESCRIPTION
**Pull request by Kleptobismol**
Deprecated `skbio.util.create_dir`
Deprecated `skbio.util.remove_files`
Deprecated `skbio.util.is_casava_v180_or_later` and moved functionality to FASTQ Sniffer
Fixes #833

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,14 @@
 
 ### Bug fixes
 
+### Deprecated functionality [experimental]
+* Deprecated function `skbio.util.create_dir`. This function will be removed in scikit-bio 0.4.4. Please use standard Python OS functionality
+described here (https://docs.python.org/2/library/os.html#os.makedirs). ([#833](https://github.com/biocore/scikit-bio/issues/833))
```
Comment (ebolyen) [inline CHANGELOG.md]: Make `here.` the link. Same for below

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,14 @@
 
 ### Bug fixes
 
+### Deprecated functionality [experimental]
+* Deprecated function `skbio.util.create_dir`. This function will be removed in scikit-bio 0.4.4. Please use standard Python OS functionality
```
Comment (ebolyen) [inline CHANGELOG.md]: `standard Python OS`  -> `the Python standard library`

File: skbio/io/format/fastq.py [hunk 3]
```diff
@@ -317,7 +317,11 @@ def _fastq_sniffer(fh):
 
     try:
         not_empty = False
-        for _ in zip(range(10), _fastq_to_generator(fh, phred_offset=33)):
+        for _, seq in zip(range(10), _fastq_to_generator(fh, phred_offset=33)):
```
Comment (ebolyen) [inline skbio/io/format/fastq.py]: I think this belongs in a different PR?

File: skbio/io/format/tests/data/fastq_single_seq_illumina1.8 [hunk 4]
```diff
@@ -0,0 +1,4 @@
+@EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG
```
Comment (ebolyen) [inline skbio/io/format/tests/data/fastq_single_seq_illumina1.8]: Same here, if you need help cleaning up your `git` history, come see me or @jairideout later this afternoon.

File: skbio/io/format/tests/test_fastq.py [hunk 5]
```diff
@@ -138,6 +138,11 @@ def test_negatives(self):
         for fp in self.negatives:
             self.assertEqual(_fastq_sniffer(fp), (False, {}))
 
+    def test_illumina_sniffed(self):
```
Comment (ebolyen) [inline skbio/io/format/tests/test_fastq.py]: unrelated

File: skbio/io/format/fastq.py (hunk 3 above)
Comment (ebolyen) [inline skbio/io/format/fastq.py]: Oh my bad, I see the `is_casava_v180_or_later` function.

File: skbio/util/_misc.py [hunk 6]
```diff
@@ -226,7 +229,9 @@ def safe_md5(open_file, block_size=2 ** 20):
     return md5
 
 
-@experimental(as_of="0.4.0")
+@deprecated(as_of="0.4.2-dev", until="0.4.4",
+            reason="Deprecated in favor of solutions present in standard "
+            "python library")
```
Comment (ebolyen) [inline skbio/util/_misc.py]: Python standard library

File: skbio/util/_misc.py [hunk 7]
```diff
@@ -268,7 +273,9 @@ def remove_files(list_of_filepaths, error_on_missing=True):
                       '\t'.join(missing))
 
 
-@experimental(as_of="0.4.0")
+@deprecated(as_of="0.4.2-dev", until="0.4.4",
+            reason="Deprecated in favor of solutions present in standard "
+            "python library")
```
Comment (ebolyen) [inline skbio/util/_misc.py]: Python standard library

File: skbio/util/_misc.py [hunk 8]
```diff
@@ -226,7 +229,9 @@ def safe_md5(open_file, block_size=2 ** 20):
     return md5
 
 
-@experimental(as_of="0.4.0")
+@deprecated(as_of="0.4.2-dev", until="0.4.4",
```
Comment (ebolyen) [inline skbio/util/_misc.py]: `until` is a little tricky here, our next release is *hopefully* going to be `0.5.0`, @jairideout how do you want to handle this?

File: skbio/io/format/fastq.py [hunk 9]
```diff
@@ -317,7 +317,11 @@ def _fastq_sniffer(fh):
...
+            if len((seq.metadata['id'] +
+                    seq.metadata['description']).split(':')) == 10 and \
+               seq.metadata['description'].split(':')[1] in 'YN':
```
Comment (ebolyen) [inline skbio/io/format/fastq.py]: This code is a little messy (the indentation especially, though you have satisfied PEP8!). It might be worth putting some temp variables in instead of making it one long expression.

Comment (Kleptobismol) [inline skbio/io/format/fastq.py]: How's this?
```python
split_description = seq.metadata['description'].split(':')
split_length = len(seq.metadata['id'] + split_description)
if split_length == 10 and split_description[1] in 'YN':
...
```

Comment (ebolyen) [inline skbio/io/format/fastq.py]: :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (Kleptobismol) [inline CHANGELOG.md]: By here do you mean [this](https://github.com/biocore/scikit-bio/pull/1292)?

Comment (Kleptobismol) [inline CHANGELOG.md]: So [here](https://docs.python.org/2/library/os.html#os.makedirs) like this.

Comment (ebolyen) [inline CHANGELOG.md]: yup!

File: skbio/util/_misc.py (hunk 7 above)
Comment (Kleptobismol) [inline skbio/util/_misc.py]: Fixed

File: skbio/util/_misc.py (hunk 8 above)
Comment (jairideout) [inline skbio/util/_misc.py]: Let's go with 0.5.1 since 0.5.0 is tentatively the next release. Sorry for giving you the wrong `until` @Kleptobismol

**Comment by ebolyen** [comment]
Thanks @Kleptobismol!
